### PR TITLE
Update LPC8xx dependencies and SVDs, re-generate all crates

### PR DESCRIPTION
**Pull request by hannobraun**
I didn't mean for this pull request to get this big, but one thing kind of led to another here. As the title says, this is mostly about the LPC8xx crates, but since I integrated those into `generate.sh`, I had to re-generate everything else, too.

I hope I'm not overstepping my bounds here, but I want to merge this today. I'm giving an embedded workshop tomorrow, and I'd like to avoid having to use Git dependencies for that. I also don't want to use outdated crates, hence my urgency.

I think most of this should be uncontroversial, except for on thing: I've integrated `lpc845-pac` into `generate.sh`, even though @david-sawatzke previously stated that he prefers the patch-based approach. I personally found the patch-based to be tedious, but  I don't know if the Git-based approach is actually better. Time will tell.

I guess the only case I can really make is that this update was a lot of work, committing the SVD files to the repository and modifying them there seemed a lot easier, and in the end I was the one doing the work, so I got to decide. I hope that's fine for now. I don't mean to kill any ongoing discussions about this.

I've ported `lpc8xx-hal` (https://github.com/lpc-rs/lpc8xx-hal/pull/160). As far as I can tell, things mostly still work.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*/svd2rust-warnings.log
```
Comment (rnestler) [inline .gitignore]: Missing newline at end of file.

**Review by rnestler — COMMENTED** [review]
Just a minor nitpick. I didn't test any of the generated code. Maybe I'll find time next week to test the lpc11uxx crate a bit.

File: .gitignore (hunk 1 above)
Comment (hannobraun) [reply .gitignore]: Thanks, fixed.

**Comment by hannobraun** [comment]
Merging now. Again, I hope I'm not offending anyone by merging this so quickly. I'd be happy to clean up any unintended fallout after my workshop!

**Pull request merged**